CourseGrab/LoginViewController.swift

```diff
@@ -2,21 +2,99 @@
 //  LoginViewController.swift
 //  CourseGrab
 //
-//  Created by Daniel Vebman on 1/26/20.
+//  Created by Reade Plunkett on 1/26/20.
 //  Copyright © 2020 Cornell AppDev. All rights reserved.
 //
 
+import GoogleSignIn
+import SnapKit
 import UIKit
 
-class LoginViewController: UIViewController {
+class LoginViewController: UIViewController, GIDSignInDelegate {
+
+    private let stackView = UIStackView()
+    private let titleLabel = UILabel()
+    private let logoView = UIImageView()
+    private let signInButton = GIDSignInButton()
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
 
         view.backgroundColor = .white
+
+        GIDSignIn.sharedInstance().delegate = self
+        GIDSignIn.sharedInstance()?.presentingViewController = self
+        // Automatically sign in the user.
+        GIDSignIn.sharedInstance()?.restorePreviousSignIn()
+
+        setUpStackView()
+        setUpTitleLabel()
+        setUpLogoView()
+        setUpSignInButton()
+
+    }
+
+    private func setUpStackView() {
+        stackView.axis = .vertical
+        stackView.distribution = .fill
+        stackView.alignment = .center
+        stackView.spacing = 40
+        view.addSubview(stackView)
+
+        stackView.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(32)
+            make.centerY.equalToSuperview().priority(.high)
+        }
+    }
+
+    private func setUpTitleLabel() {
+        titleLabel.text = "CourseGrab"
+        titleLabel.textColor = .black
+        titleLabel.textAlignment = .center
+        titleLabel.font = .systemFont(ofSize: 36, weight: .bold)
+        stackView.addArrangedSubview(titleLabel)
     }
 
+    private func setUpLogoView() {
+        logoView.image = UIImage(named: "image")
+        logoView.contentMode = .scaleAspectFit
+        stackView.addArrangedSubview(logoView)
+
+        logoView.snp.makeConstraints { make in
+            make.height.equalTo(128)
+        }
+    }
+
+    private func setUpSignInButton() {
+        stackView.addArrangedSubview(signInButton)
+    }
+
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error!) {
+        if let error = error {
+            if (error as NSError).code == GIDSignInErrorCode.hasNoAuthInKeychain.rawValue {
+                // User is not signed in or have signed out
+            } else {
+                //print("\(error.localizedDescription)")
+            }
+            return
+        }
+        let coursesViewController = CoursesViewController()
+        let coursesNavigationController = CoursesNavigationController(rootViewController: coursesViewController)
+        view.window?.rootViewController = coursesNavigationController
```

Comment:
others can give their input but I think its better if we access the `window` by accessing the shared `UIApplicationDelegate` and change the `rootViewController` in that way

Reply:
I think it would make more sense to avoid the window in general and simply present the navigation controller full screen: 

    coursesNavigationController.modalPresentationStyle = .fullScreen
    present(navigationController, animated: false)